signer/src/stacks/api.rs

```diff
@@ -1458,31 +1305,34 @@ where
         // There are more blocks to fetch, so let's get them. This assumes
         // optimistically that the parent is still a Nakamoto block (and so has
         // a tenure); if that's not the case, we get an `Err` here.
-        let tenure_headers_result = stacks.get_tenure_headers(&header.parent_block_id).await;
+        let tenure_headers_result = stacks.get_tenure_headers(anchor_block_height - 1).await;
+        anchor_block_height = anchor_block_height - 1;
         let tenure_headers = match tenure_headers_result {
             Ok(tenure_headers) => tenure_headers,
             Err(error) => {
-                // A 404 could mean that we reached the Nakamoto start height
-                // and we tried fetching a tenure for a pre-Nakamoto block
-                if stacks
-                    .check_pre_nakamoto_block(&header.parent_block_id)
-                    .await
-                    .is_ok()
-                {
-                    tracing::debug!(
-                        %nakamoto_start_height,
-                        last_chain_length = %tenure.anchor_block_height,
-                        "all Nakamoto blocks fetched; stopping"
-                    );
-                    break;
-                }
-                return Err(error);
+                tracing::warn!(
+                    %error,
+                    "error during fetching tenure headers"
+                );
+                continue;
```

Comment:
I think we should check the error and `continue` only if it's the "tenure not found" error. If we get some issue with the stacks node (eg, it goes down) while in this loop we will keep iterating until we reach the nakamoto height.
We would also keep `db.write_stacks_block_headers` the same tenure over and over, which is fine because of the `ON CONFLICT DO NOTHING` but maybe we can avoid it? Also note that the in memory implementation of `write_stacks_block_headers` is not idempotent